app/Services/EDocument/Imports/ZugferdEDocument.php

```diff
@@ -0,0 +1,75 @@
+
+use App\Factory\ExpenseFactory;
+use App\Models\Currency;
+use App\Repositories\VendorRepository;
+use App\Services\AbstractService;
+use Exception;
+use horstoeko\zugferd\ZugferdDocumentReader;
+
+class ZugferdEDocument extends AbstractService
+{
+    public ZugferdDocumentReader $document;
+
+    /**
+     * @throws Exception
+     */
+    public function __construct(public string $tempdocument)
+    {
+        $this->document = ZugferdDocumentReader::readAndGuessFromContent($this->tempdocument);
+        # curl -X POST http://localhost:8000/api/v1/edocument/upload -H "Content-Type: multipart/form-data" -H "X-API-TOKEN: 7tdDdkz987H3AYIWhNGXy8jTjJIoDhkAclCDLE26cTCj1KYX7EBHC66VEitJwWhn" -H "X-Requested-With: XMLHttpRequest" -F _method=PUT -F documents[]=@einvoice.xml
+    }
+
+    /**
+     * @throws Exception
+     */
+    public function run(): string
+    {
+        $user = auth()->user();
+
+        $expense = ExpenseFactory::create($user->company()->id, $user->id);
+
+        //
+        $this->document->getDocumentInformation($documentno, $documenttypecode, $documentdate, $invoiceCurrency, $taxCurrency, $documentname, $documentlanguage, $effectiveSpecifiedPeriod);
+        nlog("Profile:               {$this->document->getProfileDefinitionParameter("name")}\r\n");
+        nlog("Profile:               {$this->document->getProfileDefinitionParameter("altname")}\r\n");
+        nlog("Document Type:         $documenttypecode\r\n");
+        $expense->date = $documentdate;
+        $expense->user_id = $user->id;
+        $expense->company_id = $user->company()->id;
+        $expense->public_notes = $documentno;
+        $expense->currency_id = Currency::whereCode($invoiceCurrency);
+        if ($taxCurrency != $invoiceCurrency){
+            nlog("Unusal case in import happend: InvoiceCurrency unequal to taxcurrency");
+        }
+        $expense->uses_inclusive_taxes = false;
+
+        $this->document->getDocumentSummation($grandTotalAmount, $duePayableAmount, $lineTotalAmount, $chargeTotalAmount, $allowanceTotalAmount, $taxBasisTotalAmount, $taxTotalAmount, $roundingAmount, $totalPrepaidAmount);
+        $expense->amount = $lineTotalAmount;
+        $counter = 0;
+        if ($this->document->firstDocumentTax()) {
+            $counter += 1;
+            do {
+                $this->document->getDocumentTax($categoryCode, $typeCode, $basisAmount, $calculatedAmount, $rateApplicablePercent, $exemptionReason, $exemptionReasonCode, $lineTotalBasisAmount, $allowanceChargeBasisAmount, $taxPointDate, $dueDateTypeCode);
+                $expense->${"tax_amount$counter"} = $calculatedAmount;
+                $expense->${"tax_rate$counter"} = $rateApplicablePercent;
+            } while ($this->document->nextDocumentTax());
+        }
+        // TODO find vendor
+        $vendors_registration = VendorRepository::class;
+$vendor = $vendors_registration->findVendorByNumber($this->document->getDocumentSeller()->getGlobalID());
+        return "";
```

Comment:
@turbo124 when creating an expense through the api the following events are thrown:

```
event(new ExpenseWasCreated($expense, $expense->company, Ninja::eventVars(null))); // @turbo124 please check, I copied from API-Controller
event('eloquent.created: App\Models\Expense', $expense); // @turbo124 please check, I copied from API-Controller
```

should they also be present here?